src/com/lightning/jpipeworks/things/RecoloringSpace.java

```diff
@@ -0,0 +1,66 @@
+package com.lightning.jpipeworks.things;
+
+import java.util.Objects;
+import java.util.function.IntUnaryOperator;
+
+public class RecoloringSpace implements DrawingSpace {
```

Comment:
This is a good idea for when we:
1. Have a fully working resource system where having different spaces for the same graphics works
2. Have a decent engine.